spacepy/pybats/__init__.py

```diff
@@ -1954,36 +2025,74 @@ def write(self, outfile=False):
+                    out.write('#POSITION\n{0[1]:-7.3f}\n{0[2]:-7.3f}\n\n'
                               .format(xyz).encode())
                 elif None not in xyz:
                     out.write('#SATELLITEXYZ\n{}\n'.format(
-                        ''.join("{:-6.2f}\n".format(n) for n in xyz)).encode())
+                        ''.join("{:-7.3f}\n".format(n) for n in xyz)).encode())
             if self.attrs['delay']:
                 out.write('#DELAY\n{:-9.2f}\n\n'.format(self.attrs['delay'])
                           .encode())
             if None not in self.attrs['plane']:
                 out.write('#PLANE\n{}\n'.format(
-                    ''.join('{:-6.2f}\n'.format(n)
+                    ''.join('{:-7.3f}\n'.format(n)
                             for n in self.attrs['plane'])).encode())
 
             # Write the data:
             out.write(b'\n#START\n')
             # Round time to millisecond and format it
             timestr = np.vectorize(
-                lambda t: (t.replace(microsecond=0)
-                + dt.timedelta(microseconds=int(round(t.microsecond, -3))))
+                lambda t: (t.replace(microsecond=0) +
+                           dt.timedelta(microseconds=int(round(
+                               t.microsecond, -3))))
                 .strftime('%Y %m %d %H %M %S %f')[:-3] + ' ',
                 otypes=[bytes])(self['time'])
             outarray = np.column_stack([timestr] + [
                     np.char.mod('%10.2f', self[key]) for key in var])
             np.savetxt(out, outarray, delimiter=' ', fmt='%s')
 
+    def _set_attributes(self):
+        '''
+        Set attributes, including units and axes labels.
+        Units should use LaTeX formatting.
+        Labels should be the variable name in human readable format
+        using LaTeX as necessary.
+
+        Plotting functions combine the label and units for y-axis labels
+        and the label for axes legend uses.
+        '''
+
+        # Vector quantities:
+        for x in 'xyz':
+            # Magnetic field:
+            self['b'+x].attrs['units'] = '$nT$'
+            self['b'+x].attrs['label'] = f'IMF B$_{x.upper()}$'
+
+            # Bulk velocity:
+            self['u'+x].attrs['units'] = '$km/s$'
+            self['u'+x].attrs['label'] = f'U$_{x.upper()}$'
+
+        # Densities & temperature:
+        for v in self.keys():
+            if v[0]=='b' or v[0]=='u': continue
+            if v == 't':
+                self[v].attrs['units'] = '$K$'
+                self[v].attrs['label'] = "T"
+            elif 'rho' in v.lower():
+                self[v].attrs['units'] = r'$cm^{-3}$'
+                self[v].attrs['label'] = r'$\rho_{'+v[:-3]+r'}$'
+            elif v=='n':
+                self[v].attrs['units'] = r'$cm^{-3}$'
+                self[v].attrs['label'] = r'$\rho$'
+            else:
+                self[v].attrs['units'] = ''
+                self[v].attrs['label'] = v
+                
```

Comment:
Rebased. Second definition deleted because the first was edited later on, so this now looks like a move of the function in "fixed tests and addressed bugs."